contracts/SwapFlashLoan.sol

```diff
@@ -0,0 +1,169 @@
+// SPDX-License-Identifier: MIT WITH AGPL-3.0-only
+
+pragma solidity 0.6.12;
+
+import "./Swap.sol";
+import "./interfaces/IFlashLoanReceiver.sol";
+
+/**
+ * @title Swap - A StableSwap implementation in solidity.
+ * @notice This contract is responsible for custody of closely pegged assets (eg. group of stablecoins)
+ * and automatic market making system. Users become an LP (Liquidity Provider) by depositing their tokens
+ * in desired ratios for an exchange of the pool token that represents their share of the pool.
+ * Users can burn pool tokens and withdraw their share of token(s).
+ *
+ * Each time a swap between the pooled tokens happens, a set fee incurs which effectively gets
+ * distributed to the LPs.
+ *
+ * In case of emergencies, admin can pause additional deposits, swaps, or single-asset withdraws - which
+ * stops the ratio of the tokens in the pool from changing.
+ * Users can always withdraw their tokens via multi-asset withdraws.
+ *
+ * @dev Most of the logic is stored as a library `SwapUtils` for the sake of reducing contract's
+ * deployment size.
+ */
+contract SwapFlashLoan is Swap {
+    // Fee parameters used to calculate fees for flashloans.
+    uint256 public flashLoanFeeBips;
+    uint256 public flashLoanProtocolFeeBips;
```

Comment:
despite following the same naming convention these variables have different semantics. I would conder renaming `flashLoanProtocolFeeBips` to `protocolFeeShareBips` and adding a comment explaining that it's the portion of the total flashloan fee that is allocated to the protocol rather than the pool